data/users/keystone.py

```diff
@@ -137,7 +137,7 @@ def verify_credentials(self, username_or_email, password):
             return (None, "Missing email field for user %s" % user_id)
 
         email = user.email if hasattr(user, "email") else None
-        return (UserInformation(username=username_or_email, email=email, id=user_id), None)
+        return (UserInformation(username=user.name, email=email, id=user_id), None)
```

Comment:
And `name` is guaranteed to be present?

Reply:
Yes, see https://sergslipushenko.github.io/html/api-ref-identity-admin-v2.html

Reply:
Specifically below:

> A `user` object, which shows the `username`, `roles_links`, `id`, `roles`, and `name`.

Reply:
Shouldn't this be `username` then?

Reply:
Honestly, I didn't investigate harder to figure out what's the difference between `username` and `name`. This is just being consistent with the field keystonev3 uses(though appears `username` isn't  a field anymore).

Reply:
Can you run a test and verify?

Reply:
Yeah, everything looks good!